src/App.tsx

```diff
@@ -1,223 +1,272 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
 import { AuthContext } from './components/Auth/AuthContext';
+import {
+  addTodo,
+  changeTodo,
+  getTodos,
+  removeTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { Types } from './types/Types';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
+import { NewTodo } from './components/NewTodo/NewTodo';
+import { ErrorNote } from './components/ErrorNote/ErrorNote';
+
+const FILTERS: Types = {
+  all: 'all',
+  completed: 'completed',
+  active: 'active',
+};
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState(FILTERS.all);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [errorStatus, setErrorStatus] = useState(false);
+  const [isAdding, setIsAdding] = useState(false);
+  const [isRemoving, setIsRemoving] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo>({
+    id: 0,
+    title: '',
+    userId: 0,
+    completed: false,
+  });
+
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const changeErrorStatus = () => setErrorStatus(false);
+
+  async function loadTodos() {
+    if (user) {
+      const todosFromServer = await getTodos(user.id);
+
+      setTodos(todosFromServer);
+    }
+  }
+
+  const addTodoOnServer = async (todoTitle: string) => {
+    if (user) {
+      try {
+        setTempTodo({
+          id: 0,
+          title: todoTitle,
+          userId: user.id,
+          completed: false,
+        });
+        setIsAdding(true);
+
+        const newTodo = await addTodo({
+          title: todoTitle,
+          userId: user.id,
+          completed: false,
+        });
+
+        setTodos(currTodos => [...currTodos, newTodo]);
+        setIsAdding(false);
+      } catch {
+        setErrorMessage('Unable to add a todo');
+        setErrorStatus(true);
+        setIsAdding(false);
+        setTempTodo({
+          id: 0,
+          title: '',
+          userId: 0,
+          completed: false,
+        });
+      }
+    }
+  };
+
+  const removeTodoFromServer = async (todoId: number) => {
+    try {
+      setIsRemoving(prevIds => [...prevIds, todoId]);
+      await removeTodo(todoId);
+
+      setTodos(currTodos => currTodos.filter(todo => todo.id !== todoId));
+      setIsRemoving(prevIds => prevIds.filter(id => id !== todoId));
+    } catch {
+      setErrorMessage('Unable to delete a todo');
+      setErrorStatus(true);
+    }
+  };
+
+  const removeCompletedTodos = async () => {
+    try {
+      await Promise.all(todos.map(todo => {
```

Comment:
if you don't need the result of the map method, use for each instead of it

Reply:
unfortunately, it is showing an error when I try to use forEach with promise.all as it expects an array as an argument, tried to find a solution and didn't find any(